File: android/src/main/java/org/ligi/passandroid/AppComponent.java
package org.ligi.passandroid;

import dagger.Component;
import javax.inject.Singleton;
import org.ligi.passandroid.model.PassStore;
import org.ligi.passandroid.model.Settings;
import org.ligi.passandroid.ui.PassAdapter;
import org.ligi.passandroid.ui.PassAndroidActivity;
import org.ligi.passandroid.ui.PassEditActivity;
import org.ligi.passandroid.ui.PassImportActivity;
import org.ligi.passandroid.ui.PassListActivity;
import org.ligi.passandroid.ui.PassListFragment;
import org.ligi.passandroid.ui.PassMenuOptions;
import org.ligi.passandroid.ui.PassNavigationView;
import org.ligi.passandroid.ui.PassViewActivityBase;
import org.ligi.passandroid.ui.SearchPassesIntentService;
import org.ligi.passandroid.ui.TouchImageActivity;
import org.ligi.passandroid.ui.edit.PassandroidFragment;
import org.ligi.passandroid.ui.quirk_fix.USAirwaysLoadActivity;

@Singleton
@Component(modules = {AppModule.class , TrackerModule.class})
public interface AppComponent {

    void inject(PassViewActivityBase passViewActivityBase);

    void inject(PassListActivity passListActivity);

    void inject(PassEditActivity passEditActivity);

    void inject(PassandroidFragment passandroidFragment);

    void inject(PassAdapter passAdapter);

    void inject(PassImportActivity passImportActivity);

    void inject(PassMenuOptions passMenuOptions);

    void inject(SearchPassesIntentService searchPassesIntentService);

    void inject(USAirwaysLoadActivity usAirwaysLoadActivity);

    void inject(PassAndroidActivity passAndroidActivity);

    void inject(PassListFragment passListFragment);

    void inject(PassNavigationView passNavigationView);

    void inject(TouchImageActivity touchImageActivity);

    PassStore passStore();

    Tracker tracker();

    Settings settings();
}

File: android/src/main/java/org/ligi/passandroid/AppComponent.kt
package org.ligi.passandroid

import dagger.Component
import org.ligi.passandroid.model.PassStore
import org.ligi.passandroid.model.Settings
import org.ligi.passandroid.ui.*
import org.ligi.passandroid.ui.edit.FieldsEditFragment
import org.ligi.passandroid.ui.quirk_fix.USAirwaysLoadActivity
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AppModule::class, TrackerModule::class))
interface AppComponent {

    fun inject(passViewActivityBase: PassViewActivityBase)

    fun inject(passListActivity: PassListActivity)

    fun inject(passEditActivity: PassEditActivity)

    fun inject(passAdapter: PassAdapter)

    fun inject(passImportActivity: PassImportActivity)

    fun inject(passMenuOptions: PassMenuOptions)

    fun inject(searchPassesIntentService: SearchPassesIntentService)

    fun inject(usAirwaysLoadActivity: USAirwaysLoadActivity)

    fun inject(passAndroidActivity: PassAndroidActivity)

    fun inject(passListFragment: PassListFragment)

    fun inject(passNavigationView: PassNavigationView)

    fun inject(touchImageActivity: TouchImageActivity)

    fun inject(fieldsEditFragment: FieldsEditFragment)

    fun passStore(): PassStore

    fun tracker(): Tracker

    fun settings(): Settings

}
